src/validation/custom.validation.ts

```diff
@@ -78,11 +78,75 @@ export const checkDateValueIsValid = (errMsg: string, dayStr: string = "", month
 
 export const checkOptionalDate = (dayStr: string = "", monthStr: string = "", yearStr: string = "") => {
   if ( dayStr !== "" || monthStr !== "" || yearStr !== "" ) {
-    checkDateValueIsValid(ErrorMessages.INVALID_DATE, dayStr, monthStr, yearStr);
-    checkDateIsInPastOrToday(ErrorMessages.DATE_NOT_IN_PAST_OR_TODAY, dayStr, monthStr, yearStr);
+    const areDateFieldsPresent = checkDateFieldsPresent(dayStr, monthStr, yearStr);
+    if (areDateFieldsPresent) {
+      const isDateValid = checkDateValueIsValid(ErrorMessages.INVALID_DATE, dayStr, monthStr, yearStr);
+      if (isDateValid) {
+        const isDateInThePast = checkDateIsInPastOrToday(ErrorMessages.DATE_NOT_IN_PAST_OR_TODAY, dayStr, monthStr, yearStr);
+        if (isDateInThePast) {
+          checkDateIsWithinLast3Months(ErrorMessages.IDENTITY_CHECK_DATE_NOT_WITHIN_PAST_3_MONTHS, dayStr, monthStr, yearStr);
+        }
+      }
+    }
+  }
+  return true;
+};
+
+export const checkIdentityDate = (dayStr: string = "", monthStr: string = "", yearStr: string = "") => {
+  const isStartDateFields = checkStartDate(dayStr, monthStr, yearStr);
```

Comment:
It reads a bit odd that the `checkIdentityDate` function calls a `checkStartDate` function. Suggest that the `checkStartDate` function is renamed to be more generic - just `checkDate` perhaps (since it doesn't seem to have to take a 'start date' as input)?

Also, the variable `isStartDateFields` is confusing (at least to me!). How about `isDateValid` (assuming you rename that other function too)?

Reply:
Renamed this to "mandatory date" to compliment the existing optional date.